### PR TITLE
Filter events so they are only triggered when it really matters

### PR DESCRIPTION
**Pull request by randomdud3**
This adds a way to filtering events. If its called multiple times for the same event, filters will be joined by ors

Right now it breaks anything other than biter_battles, so don't merge it. But I would like to test it someday

### CODE REVIEW

**Comment by randomdud3** [comment]
@M3wM3w fixed. I had to move init to the top since the health module was loaded before anything that i could check, but it works now

**Comment by M3wM3w** [comment]
could the custom content in the health boost module be removed and simply these added to the end of the main.lua of bb? would it work?

`Event.add_event_filter(defines.events.on_entity_damaged, { filter = "type", type = "unit" })
Event.add_event_filter(defines.events.on_entity_damaged, { filter = "name", name = "rocket-silo" })`

**Comment by randomdud3** [comment]
fixed

File: maps/biter_battles_v2/main.lua [hunk 1]
```diff
@@ -140,7 +140,10 @@ Event.add(defines.events.on_robot_built_entity, on_robot_built_entity)
 Event.add(defines.events.on_tick, on_tick)
 Event.on_init(on_init)
 
+Event.add_event_filter(defines.events.on_entity_damaged, { filter = "name", name = "rocket-silo" })
+Event.add_event_filter(defines.events.on_entity_damaged, { flter = "type", type = "unit" })
```
Comment (mattNeumayer) [inline maps/biter_battles_v2/main.lua]: typo "filter"

**Pull request merged**